keps/sig-network/20181024-service-topology.md

```diff
@@ -108,17 +117,24 @@ kind: Service
 metadata:
   name: service-local
 spec:
-  topologyKeys: ["host", "zone"]
+  topologyKeys: ["kubernetes.io/hostname", "topology.kubernetes.io/zone"]
 ```
 
 
-In our example above, we will firstly try to find the backends in the same host. If no backends match, we will then try the lucky of same zone. If finally we can't find any backends in the same host or same zone, then we say the service has no satisfied backends and connections will fail.
+In our example above, we will firstly try to find the backends in the same host. If no backends match, we will then try the same zone. If finally we can't find any backends in the same host or same zone, then we say the service has no satisfied backends and connections will fail.
 
-If we configure topologyKeys as `["host", ""]`, we just do the effort to find the backends in the same host and will not fail the connection if no matched backends found.
+If we configure topologyKeys as `["kubernetes.io/hostname", ""]`, we just do the effort to find the backends in the same host and will not fail the connection if no matched backends found.
 
-### New PodLocator resource
+### Service Topology Scalability
 
-As EndpointAddress already contains nodeName field, we can build a service that will precook Pod to its topologies mapping. Then let all interested components(at least kube-proxy and kube-dns and coredns) just watch that precooked object and do necessary mapping internally. Given that we don't know which labels are topology labels, we are going to copy all node labels.
+The alpha release of Service Topology will have a simple implementation where kube-proxy and any other consumers of `topologyKeys` will determine the locality of backends
+by filtering Endpoints and Nodes in-memory from client-go's cache informer/lister. The alpha implementation of Service Topology will not account for scalabilty, however,
+the beta release of Service Topology will address scalability concerns by introducing a `PodLocator` resource. Note that the design and implementation details around `PodLocator`
+may vary based on user feedback in the alpha release.
+
+#### New PodLocator resource
+
+As `EndpointAddress` already contains the `nodeName` field, we can build a service that will preemptively map Pods to their topologies. From there, all interested components (at least kube-proxy, kube-dns, and coredns) can watch the `PodLocator` object and do necessary mappings internally. Given that we don't know which labels are topology labels, we are going to copy all node labels to `PodLocator`.
```

Comment:
Depending on timing of this vs EndpointSlice, we may not need PodLocator in kube-proxy at all

Reply:
@imroc can you update this KEP in the future based on EndpointSlice progress please?

Reply:
Already did in https://github.com/kubernetes/enhancements/pull/1161, and also implemented alpha in https://github.com/kubernetes/kubernetes/pull/72046 , both not merged :(

I think  https://github.com/kubernetes/enhancements/pull/1161 need update cuz this PR been merged, and I can rebase https://github.com/kubernetes/kubernetes/pull/72046 if you are going to review it.

Reply:
Thanks and sorry I missed those!